aws/credentials/credentials.go

```diff
@@ -234,10 +245,11 @@ func (c *Credentials) Expire() {
 // If the Credentials were forced to be expired with Expire() this will
 // reflect that override.
 func (c *Credentials) IsExpired() bool {
-	c.m.Lock()
-	defer c.m.Unlock()
+	c.m.RLock()
+	exp := c.isExpired()
+	c.m.RUnlock()
```

Comment:
would prefer this in a defer due to the potential of `isExpired` ever panicking

Reply:
added back in defers.